File: app/src/main/java/com/example/patiobalmax/ui/MapaEstacionamiento.java
package com.example.patiobalmax.ui;

import android.os.Bundle;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.patiobalmax.adapter.PatioAdapter;
import com.example.patiobalmax.database.AppDatabase;
import com.example.patiobalmax.database.PatioEntity;

import java.util.List;

public class MapaEstacionamiento extends AppCompatActivity {

    private ImageView ivMapa;
    private TextView tvTitulo;
    private RecyclerView rvPatios;
    private PatioAdapter patioAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_mapa_estacionamiento);

        ivMapa = findViewById(R.id.ivMapa);
        tvTitulo = findViewById(R.id.tvTitulo);
        rvPatios = findViewById(R.id.rvPatios);

        tvTitulo.setText(getResources().getString(R.string.lbl_bienvenida));

        AppDatabase db = AppDatabase.getDatabase(this);
        List<PatioEntity> patios = db.patioDao().getAll();

        patioAdapter = new PatioAdapter(patios, this::navegarAPatio);
        rvPatios.setLayoutManager(new LinearLayoutManager(this));
        rvPatios.setAdapter(patioAdapter);
    }

    private void navegarAPatio(PatioEntity patio) {
        Intent intent = new Intent(MapaEstacionamiento.this, PatioEstacionamiento.class);
        intent.putExtra("patio", patio);
        startActivity(intent);
    }
}

File: app/src/main/kotlin/com/example/patiobalmax/MapaEstacionamiento.kt
package com.example.patiobalmax

import android.content.Intent
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MapaEstacionamiento : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_mapa_estacionamiento)

        val tvTituloPatio = findViewById<TextView>(R.id.tvTituloPatio)
        tvTituloPatio.text = "Seleccione un Patio"

        findViewById<Button>(R.id.btnPatio1).setOnClickListener {
            val intent = Intent(this, PatioEstacionamiento::class.java).apply {
                putExtra("patio_numero", 1)
            }
            startActivity(intent)
        }

        findViewById<Button>(R.id.btnPatio2).setOnClickListener {
            val intent = Intent(this, PatioEstacionamiento::class.java).apply {
                putExtra("patio_numero", 2)
            }
            startActivity(intent)
        }

        findViewById<Button>(R.id.btnAdministrarUsuarios).setOnClickListener {
            startActivity(Intent(this, AdministrarUsuarios::class.java))
        }
    }
}
